lib/modules/manager/github-actions/extract.ts

```diff
@@ -86,6 +87,29 @@ function extractContainer(container: unknown): PackageDependency | undefined {
   return undefined;
 }
 
+function extractRunners(runner: unknown): PackageDependency[] {
+  const runners = [];
+  if (is.string(runner)) {
+    runners.push(runner);
+  } else if (is.array(runner, is.string)) {
+    runners.push(...runner);
+  }
+
+  return runners.map((rnr) => {
+    const [depName, ...version] = rnr.split('-');
```

Comment:
use regex and filter expressions, they can be matrix job references.

Reply:
I don't fully understand what you want to achieve, but I will try..

Reply:
@PeterNitsche `runs-on: ${{ matrix.foo-bar }}` is also possible, and that would parse wrongly with this simple split.

https://github.com/TWiStErRob/detekt-4165/blob/main/.github/workflows/CI.yml#L8
I guess this is not expected to be supported, just an example the complexity.